### DISCUSSION TITLE
File homebrew.mxcl.nginx.plist is being destroyed after restarting a service...?

### DISCUSSION THREAD

**Discussion by ahoiroman** [Homebrew/discussions#2549] [Everyday usage]
Hello everybody,

I am using laravel/valet to set up Nginx and PHP for my local dev machine.

Now I mentioned that NGINX is not coming up at all. In the Nginx-logs, I can see this:

> 2021/11/27 20:16:28 [emerg] 98117#0: unexpected end of parameter, expecting ";" in command line

After hours of searching, I found this in `/Library/LaunchDaemons/homebrew.mxcl.nginx.plist`:

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
        <key>Label</key>
        <string>homebrew.mxcl.nginx</string>
        <key>ProgramArguments</key>
        <array>
                <string>/opt/homebrew/opt/nginx/bin/nginx</string>
                <string>-g</string>
                <string>&#39;daemon off;&#39;</string>
        </array>
        <key>RunAtLoad</key>
        <true/>
        <key>WorkingDirectory</key>
        <string>/opt/homebrew</string>
</dict>
</plist>
```

Yeah, `<string>&#39;daemon off;&#39;</string>` seems to be a little funny. Well - I tried to fix that and saved the new value `<string>daemon off;</string>`. But after trying to restart the service, the faulty version was in place again. Of course I thought I made a mistake and saved it again, made sure, it was being saved correctly and then I restarted the service. Well - again it was `<string>&#39;daemon off;&#39;</string>`.

So my question: How does this faulty stuff get there? Which process, config, ... is the next thing I'd have to look at?

**Comment by ahoiroman** [comment]
Unfortunately I can not open an issue as I always get a blank page on submitting.

Well: The problem is also happening after reinstalling brew and installing nginx manually:

```
brew doctor
Your system is ready to brew.
```

```
HOMEBREW_VERSION: 3.3.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 354718c40a4e7cc4e5d1d4b03fdce8b992e04e8a
Last commit: 5 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 5a13afe71e5d31bf8a99f35531dfe29e82603af9
Core tap last commit: 26 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 13.0.0 build 1300
Git: 2.30.1 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.0.1-arm64
CLT: 13.1.0.0.1.1633545042
Xcode: N/A
Rosetta 2: false
```

```
brew update
brew install ngxing
sudo brew services start nginx
```


The result: 

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
        <key>Label</key>
        <string>homebrew.mxcl.nginx</string>
        <key>ProgramArguments</key>
        <array>
                <string>/opt/homebrew/opt/nginx/bin/nginx</string>
                <string>-g</string>
                <string>&#39;daemon off;&#39;</string>
        </array>
        <key>RunAtLoad</key>
        <true/>
        <key>WorkingDirectory</key>
        <string>/opt/homebrew</string>
</dict>
</plist>
```

**Comment by SMillerDev** [comment]
It gets copied from `/opt/homebrew/opt/nginx/Homebrew.nginx.plist` on every restart.

**Reply by ahoiroman** [reply]
Could be related to this PR: https://github.com/Homebrew/homebrew-core/pull/90023

A lot of users notice that Nginx won't come up on a new installation or upgrade.

**Comment by carlocab** [comment]
Try `brew update && brew reinstall nginx`.
